target/linux/ath79/patches-4.14/0007-irqchip-irq-ath79-intc-add-irq-cascade-driver-for-QC.patch

```diff
@@ -95,27 +104,56 @@ Signed-off-by: John Crispin <john@phrozen.org>
 +	.map = ath79_intc_map,
 +};
 +
-+static int __init qca9556_intc_of_init(
++static int __init ath79_intc_of_init(
 +	struct device_node *node, struct device_node *parent)
 +{
 +	struct irq_domain *domain;
 +	struct ath79_intc *intc;
-+	int cnt, i;
++	int cnt, i, err;
 +
-+	cnt = of_property_count_u32_elems(node, "qcom,pending-bits");
++	cnt = of_property_count_u32_elems(node, "qca,pending-bits");
 +	if (cnt > ATH79_MAX_INTC_CASCADE)
 +		panic("Too many INTC pending bits\n");
 +
 +	intc = kzalloc(sizeof(*intc), GFP_KERNEL);
 +	if (!intc)
 +		panic("Failed to allocate INTC memory\n");
++	intc->chip = dummy_irq_chip;
 +	intc->chip.name = "INTC";
-+	intc->chip.irq_unmask = ath79_intc_irq_unmask,
-+	intc->chip.irq_mask = ath79_intc_irq_mask,
++	intc->chip.irq_disable = ath79_intc_irq_disable;
++	intc->chip.irq_enable = ath79_intc_irq_enable;
 +
-+	of_property_read_u32_array(node, "qcom,pending-bits", intc->irq_mask, cnt);
-+	for (i = 0; i < cnt; i++)
++	if (of_property_read_u32(node, "qca,int-status-addr", &intc->int_status) < 0) {
++		panic("Missing address of interrupt status register\n");
++	}
++
++	of_property_read_u32_array(node, "qca,pending-bits", intc->irq_mask, cnt);
++	for (i = 0; i < cnt; i++) {
 +		intc->pending_mask |= intc->irq_mask[i];
++		intc->irq_wb_chan[i] = 0xffffffff;
++	}
++
++	cnt = of_count_phandle_with_args(
```

Comment:
Here is the bug that @lucize mentioned...
If someone does not set "qca,ddr-wb-channelXXX" properties, this value is zero and 
`domain = irq_domain_add_linear(node, cnt, &ath79_irq_domain_ops, intc);`
the second param cnt (=hwirq_max) is also zero.

Reply:
Thanks for pointing out this!